src/java.base/share/classes/java/io/InputStream.java

```diff
@@ -53,8 +54,26 @@ public abstract class InputStream implements Closeable {
     // use when skipping.
     private static final int MAX_SKIP_BUFFER_SIZE = 2048;
 
+    private static final int MIN_SKIP_BUFFER_SIZE = 128;
+
     private static final int DEFAULT_BUFFER_SIZE = 8192;
 
+    /** Skip buffer, null until allocated */
+    private byte[] skipBuffer = null;
```

Comment:
This can be removed too as you have the soft reference already.

Reply:
@liach done.